Framework/Algorithms/src/RenameWorkspaces.cpp

```diff
@@ -86,6 +96,18 @@ void RenameWorkspaces::exec() {
     }
   }
 
+  // Check all names are not used already before starting rename
+  for (size_t i = 0; i < nWs; ++i) {
```

Comment:
I think here you could use a range-based loop: `for (const auto &name : newWsName) { ...`